horovod/mxnet/partition_trainer/pos_trainer.py

```diff
@@ -0,0 +1,174 @@
+from horovod.mxnet.mpi_ops import allreduce, allreduce_
+from horovod.mxnet.mpi_ops import broadcast, broadcast_
+from horovod.mxnet.mpi_ops import init, shutdown
+from horovod.mxnet.mpi_ops import size, local_size,  rank, local_rank
+
+import mxnet as mx
+from collections import OrderedDict, defaultdict
+import types
+import time
+import warnings
+from mxnet.gluon.parameter import Parameter
+
+
+class POS_Trainer(mx.gluon.Trainer):
+    def __init__(self, params, optimizer, optimizer_params=None,
+                 gradient_predivide_factor=1.0, prefix=None):
+
+        self._world_size = size()
+        self._world_rank = rank()
+
+        self._all_params = []
+        self._all_param2idx = {}
+        param_list = []
+        if isinstance(params, (dict, OrderedDict)):
+            for key in sorted(list(params.keys())):
+                param_list.append(params[key])
+            params = param_list
+        if not isinstance(params, (list, tuple)):
+            raise ValueError(
+                "First argument must be a list or dict of Parameters, " \
+                "got %s." % (type(params)))
+        for i, param in enumerate(params):
+            if not isinstance(param, Parameter):
+                raise ValueError(
+                    "First argument must be a list or dict of Parameters, " \
+                    "got list of %s." % (type(param)))
+            if param._uuid in self._all_param2idx:
+                # Shared parameters have same uuid; only need to store one of the shared versions
+                continue
+            self._all_param2idx[param._uuid] = i
+            self._all_params.append(param)
+        self._partition_params, self._param2rank = self.partition_parameters(self._all_params)
+        self._own_part = self._partition_params[self._world_rank]
+        super(POS_Trainer, self).__init__(
+            self._own_part, optimizer, optimizer_params=optimizer_params, kvstore=None)
+        self._prefix = prefix if prefix else ""
+        self._scale = gradient_predivide_factor / size()
+        self._gradient_predivide_factor = gradient_predivide_factor
+
+
+    def partition_parameters(self, params):

```

Comment:
Add leading underscore to internal methods like this one. Also, add a brief comment describing what this does.